File: 品味Java的21種設計模式/dp_java/observer/example3/Reader.java
package cn.javass.dp.observer.example3;
/**
 * 真正的读者，为了简单就描述一下姓名
 */
public class Reader implements Observer {
    /**
     * 读者的姓名
     */
    private String name;

    public void update(String content) {
        //这是采用推的方式
        System.out.println(name + "收到报纸了，阅读先。内容是===" + content);
    }
    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }
}

File: 品味Java的21種設計模式/dp_cpp/observer/example3/Reader.cpp
#include "Reader.h"

namespace cn {
namespace javass {
namespace dp {
namespace observer {
namespace example3 {

void Reader::update(std::string content) {
    //这是采用推的方式
    puts(name + "收到报纸了，阅读先。内容是===" + content);
}

std::string Reader::getName() {
    return name;
}

void Reader::setName(std::string name) {
    this->name = name;
}
}
}
}
}
}
